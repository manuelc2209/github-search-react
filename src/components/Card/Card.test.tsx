import React from "react";
import { render } from "@testing-library/react";
import { Card } from "./";

const userData = {
  login: "ads",
  id: 131052,
  node_id: "1313VXNlcjEzMTA1Mg==",
  avatar_url: "https://avatars.githubusercontent.com/u/131052?v=4",
  gravatar_id: "",
  url: "https://api.github.com/users/ads",
  html_url: "https://github.com/ads",
  followers_url: "https://api.github.com/users/ads/followers",
  following_url: "https://api.github.com/users/ads/following{/other_user}",
  gists_url: "https://api.github.com/users/ads/gists{/gist_id}",
  starred_url: "https://api.github.com/users/ads/starred{/owner}{/repo}",
  subscriptions_url: "https://api.github.com/users/ads/subscriptions",
  organizations_url: "https://api.github.com/users/ads/orgs",
  repos_url: "https://api.github.com/users/ads/repos",
  events_url: "https://api.github.com/users/ads/events{/privacy}",
  received_events_url: "https://api.github.com/users/ads/received_events",
  type: "User",
  site_admin: false,
  profileData: {
    name: "asd",
    id: 123123123123,
    company: "home",
    email: "a@email.com",
    username: "asdUsername",
    avatar_url: "http://www.google.com",
  },
  score: 1,
};

test("render a card with dummy data", () => {
  const view = render(<Card user={userData} />);
  expect(view).toMatchSnapshot();
});
