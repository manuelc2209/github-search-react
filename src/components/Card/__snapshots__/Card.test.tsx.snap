// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render a card with dummy data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-gEvEer cWcOBp"
        >
          <div
            class="sc-eqUAAy dHOlDZ"
          >
            <img
              class="sc-aXZVg hOybyn"
              src="https://avatars.githubusercontent.com/u/131052?v=4"
            />
            <div
              class="sc-iGgWBj gnmhnl"
            >
              <h5
                class="sc-fqkvVR jkfviT"
              >
                Name: 
                asd
              </h5>
              <h6
                class="sc-dcJsrY iWRpLS"
              >
                User Email: 
                a@email.com
              </h6>
              <h6>
                Username: 
                ads
              </h6>
              <p
                class="sc-gsFSXq crWHRf"
              >
                Company: 
                home
              </p>
            </div>
            <span
              class="sc-kAyceB jkPlHY"
            >
              User
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-gEvEer cWcOBp"
      >
        <div
          class="sc-eqUAAy dHOlDZ"
        >
          <img
            class="sc-aXZVg hOybyn"
            src="https://avatars.githubusercontent.com/u/131052?v=4"
          />
          <div
            class="sc-iGgWBj gnmhnl"
          >
            <h5
              class="sc-fqkvVR jkfviT"
            >
              Name: 
              asd
            </h5>
            <h6
              class="sc-dcJsrY iWRpLS"
            >
              User Email: 
              a@email.com
            </h6>
            <h6>
              Username: 
              ads
            </h6>
            <p
              class="sc-gsFSXq crWHRf"
            >
              Company: 
              home
            </p>
          </div>
          <span
            class="sc-kAyceB jkPlHY"
          >
            User
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
